File: Algorithms/C++/373-Find_K_Pairs_with_Smallest_Sums.cpp
// brute force
// Runtime: 548 ms, faster than 7.32% of C++ online submissions for Find K Pairs with Smallest Sums.
// Memory Usage: 118.3 MB, less than 20.00% of C++ online submissions for Find K Pairs with Smallest Sums.
class Solution {
public:
    vector<vector<int>> kSmallestPairs(vector<int>& nums1, vector<int>& nums2, int k) {
        vector<vector<int>> ans;
        for (int n1 : nums1)
            for (int n2 : nums2) {
                vector<int> tmp = {n1, n2};
                ans.push_back(tmp);
            }
        sort(ans.begin(), ans.end(), [](const vector<int>& num1, const vector<int>& num2) {
            return num1[0] + num1[1] < num2[0] + num2[1];
        });
        if (ans.size() <= k)
            return ans;
        ans.erase(ans.begin() + k, ans.end());
        return ans;
    }
};

File: Algorithms/Python3.x/373-Find_K_Pairs_with_Smallest_Sums.py
# Runtime: 340 ms, faster than 25.80% of Python3 online submissions for Find K Pairs with Smallest Sums.
# Memory Usage: 47.6 MB, less than 11.11% of Python3 online submissions for Find K Pairs with Smallest Sums.
class Solution:
    def kSmallestPairs(self, nums1: List[int], nums2: List[int], k: int) -> List[List[int]]:
        ans = []
        for num1 in nums1:
            for num2 in nums2:
                ans.append([num1, num2])
        return sorted(ans, key=lambda x: x[0] + x[1])[:k]


# Runtime: 320 ms, faster than 27.69% of Python3 online submissions for Find K Pairs with Smallest Sums.
# Memory Usage: 47.5 MB, less than 11.11% of Python3 online submissions for Find K Pairs with Smallest Sums.
class Solution:
    def kSmallestPairs(self, nums1: List[int], nums2: List[int], k: int) -> List[List[int]]:
        return sorted([[n1, n2] for n1 in nums1 for n2 in nums2], key=lambda x: x[0] + x[1])[:k]
